tests/unit/metadata/bucketclient/backend.js

```diff
@@ -0,0 +1,92 @@
+const assert = require('assert');
+const http = require('http');
+const BucketClientInterface =
+    require('../../../../lib/metadata/bucketclient/backend').default;
+
+const bucketName = 'test';
+const invalidBucket = 'invalid';
+const params = {};
+const logger = {
+    info: () => {},
+    debug: () => {},
+    error: () => {},
+    getSerializedUids: () => {},
+};
+const port = 9000;
+
+let bucketclient;
+let server;
+
+const dataJSON = { message: 'This is Correct!!' };
+
+function makeResponse(res, code, message, data, md) {
+    /* eslint-disable no-param-reassign */
+    res.statusCode = code;
+    res.statusMessage = message;
+    /* eslint-disable no-param-reassign */
+    if (data) {
+        res.write(JSON.stringify(data));
+    }
+    if (md) {
```

Comment:
You would have to move this block above the `if (data)` block to be safe when both `data` and `md` are set, because `setHeader` is setting an implicit header that is flushed before `write` gets called (see https://nodejs.org/api/http.html#http_response_setheader_name_value and https://nodejs.org/api/http.html#http_response_write_chunk_encoding_callback).

Reply:
Thank you. I have read the documentation, and will adjust the code as necessary.